### PR TITLE
Strip Heroku-style usernames from config url

### LINKED ISSUE #36: Handle heroku redis urls smartly
Heroku redis addon url's generally have a placeholder for the username, ie

```
redis://x:345932ywwe659weotwy@some-host:12345
```

That `x` in that url is pointless and breaks the redis package we use.  The required url is:

```
redis://:345932ywwe659weotwy@some-host:12345
```

We should make this plugin handle the url if it includes the username placeholder.

### PR DESCRIPTION
**Pull request by acorncom**
Fixes #36

### CODE REVIEW

**Comment by acorncom** [comment]
@lukemelia Thoughts here?

File: lib/redis.js [hunk 1]
```diff
@@ -211,5 +211,17 @@ module.exports = CoreObject.extend({
         }
       });
     });
+  },
+
+  _stripUsernameFromConfigUrl: function(configUrl) {
+
```
Comment (ghedamat) [inline lib/redis.js]: nitpick, can you remove this?

**Comment by ghedamat** [comment]
@acorncom thanks for this, LGTM aside from the tiny comment

just catching up on the issue here,

basically we always need to specify username separately for that to work then?

/cc @achambers @lukemelia

**Comment by lukemelia** [comment]
LGTM @acorncom. I'm a little worried that some setups may use the username somehow, but I guess we can introduce a config flag later if that turns out to be the case. Merge when you're happy @ghedamat

**Comment by acorncom** [comment]
@ghedamat Re: always needing to specify the username, from what I've been seeing it looks like the underlying `then-redis` library swaps the Redis DB into the username spot (more details here: https://github.com/ember-cli-deploy/ember-cli-deploy-redis/issues/51#issuecomment-214757164).  Seems like it's not properly following the Redis URL spec.

At this point, I'm not sure if we want to merge this PR or work on fixing the upstream library (assuming they're willing to do it).  But I'm not a huge fan of using non-spec compliant URLs ;-)

Thoughts?

**Comment by acorncom** [comment]
@ghedamat Spacing issue fixed too ;-)

**Comment by JoshSmith** [comment]
I've been lurking on this issue for awhile, and I would vote for attempting to fix the upstream library.

**Comment by ghedamat** [comment]
thanks @acorncom and sorry for the delay!

**Comment by ghedamat** [comment]
@lukemelia maybe this is worth a `minor` version bump and a mention in the changelog?

**Comment by lukemelia** [comment]
@ghedamat agreed.

**Pull request merged**